python3Packages.mathutils: init at 3.3.0

autra opened the pull request:
## Description of changes

This PR packages mathutils for nix. The motivation is mainly for #312380 and #312381 because mathutils is now a dependency.

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

kirillrdy commented:
@autra greetings !

can you split into two commits

- maintainers: add autra
- python3Packages.mathutils: init at 3.3.0 
as per https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#commit-conventions

kirillrdy commented on pkgs/development/python-modules/mathutils/default.nix:
```diff
@@ -0,0 +1,21 @@
+buildPythonPackage rec {
+  pname = "mathutils";
+  version = "3.3.0";
+  format = "setuptools";
```
```suggestion
  pyproject = true;
```

https://nixos.org/manual/nixpkgs/unstable/#buildpythonpackage-parameters

even for projects that don't use pyproject.toml it will use pypa build gooks

kirillrdy commented on pkgs/development/python-modules/mathutils/default.nix:
```diff
@@ -0,0 +1,21 @@
+  version = "3.3.0";
+  format = "setuptools";
+
+  src = fetchPypi {
```
fetching from github ( or original repo ) is prefered since packages published to pypi usually don't have tests

autra replied:
In this case, this project doesn't have tests, and the maintainer doesn't tag which commits they released. That's why I decided to fetch from pypi.

But after thinking about this, I think it's better to fetch from gitlab indeed, also because it makes easier to audit the code. After all, I have no guarantee the pypi package is shipping the code in the repository... So changing that, yes!

kirillrdy commented on pkgs/development/python-modules/mathutils/default.nix:
```diff
@@ -0,0 +1,21 @@
+  pythonImportsCheck = [ "mathutils" ];
+
+  meta = with lib; {
+    description = "Utilities for working with datetime objects.";
```
https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#meta-attributes

must not end in period

autra replied:
This was completely wrong :-O I've put the correct description.

kirillrdy commented on pkgs/development/python-modules/mathutils/default.nix:
```diff
@@ -0,0 +1,21 @@
+  meta = with lib; {
+    description = "Utilities for working with datetime objects.";
+    homepage = "https://gitlab.com/ideasman42/blender-mathutils";
+    license = licenses.gpl2;
```
gpl2 is deprecated, please use one of gpl2Only, gpl2Classpath, gpl2ClasspathPlus, etc ... see  lib/licenses.nix

autra replied:
Changed it to gpl2Plus.

autra committed:
maintainers: add autra

autra committed:
python3Packages.mathutils: init at 3.3.0

The main motivation is to fix ifcopenshell packages, because mathutils
is now a dependency.

autra commented:
@kirillrdy done! Thanks for educating me :-)

kirillrdy commented:
thank you for PR and welcome to nixpkgs !
